addons/udes_stock/models/stock_picking.py

```diff
@@ -0,0 +1,175 @@
+            and a subset of stock.move.lines are then moved into it.
+            The move from which the move lines have been transferred
+            has the ordered_qty decremented by the amount of the
+            transferred lines.
+        """
+        Move = self.env['stock.move']
+        # Based on back order creation in stock_move._action_done
+        self.ensure_one()
+
+        if mls is None:
+            mls = self.move_line_ids.filtered(lambda x: x.qty_done > 0)
+
+        # test that the intercetion of mls and move lines in picking
+        # therefore we have some relevent move lines
+        if not (mls & self.move_line_ids):
+            raise ValidationError(
+                _(f'There is no move lines within picking {self.name} to backorder')
+            )
+
+        new_moves = Move.browse()
+
+        for current_move in mls.mapped('move_id'):
+            current_mls = mls.filtered(lambda x: x.move_id == current_move)
+            new_moves |= current_move.split_out_move_lines(current_mls)
+
+        # Create picking for completed move lines
+        bk_picking = self.copy(
+            {'name': '/', 'move_lines': [], 'move_line_ids': [], 'backorder_id': self.id,}
+        )
+        new_moves.write({'picking_id': bk_picking.id})
+        new_moves.mapped('move_line_ids').write({'picking_id': bk_picking.id})
+
+        return bk_picking
+
+    def _requires_backorder(self, mls):
+        """ Checks if a backorder is required by checking if all move.lines
+            within a picking are present in mls
+            TODO: Consolidate with _check_backorder in Odoo core - figure out cases _check_backorder misses
+        """
+        mls_moves = mls.mapped('move_id')
+        for move in self.move_lines:
+            if (
+                move not in mls_moves
+                or not move.move_line_ids == mls.filtered(lambda x: x.move_id == move)
+                or move.move_orig_ids.filtered(lambda x: x.state not in ('done', 'cancel'))
+            ):
+                return True
+        return False
+
+    def create_picking(
+        self,
+        picking_type,
+        products_info,
+        location_id=None,
+        location_dest_id=None,
+        confirm=False,
+        assign=False,
+        create_batch=False,
+        **kwargs,
+    ):
+        """ Create and return a picking for the given picking_type
+            Function based on test function in Odoo11, with some notions
+            retained from craet_picking in udes_stock
+            :args:
+                - picking_type
+                - products_info
+            :kwargs:
+                - location_id
+                - location_dest
+                - confirm
+                - asign
+                - create_batch
+        """
+        Picking = self.env['stock.picking']
+        # if location_id not specified, revert to default for that picking_type
+        if location_id is None:
```

Comment:
Remove the two ifs related to locations. By default pick them from picking type, and if they are passed that will be picked up by `vals.update(kwargs))`